wormhole-connect/src/hooks/useBalanceChecker.ts

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState, useMemo } from 'react';
+import { amount, routes } from '@wormhole-foundation/sdk';
+import { getWalletConnection, TransferWallet } from 'utils/wallet';
+import config from 'config';
+import { useIsMounted } from './useIsMounted';
+
+export const useBalanceChecker = (
+  quote?: routes.Quote<
+    routes.Options,
+    routes.ValidatedTransferParams<routes.Options>
+  >,
+): {
+  feeSymbol?: string;
+  isCheckingBalance: boolean;
+  hasSufficientBalance: boolean;
+  walletBalance?: number;
+  networkCost?: number;
+} => {
+  const [isCheckingBalance, setCheckingBalance] = useState(false);
+  const [state, setState] = useState<{ balance: number; cost: number } | null>(
+    null,
+  );
+  const isMounted = useIsMounted();
+
+  const feeSymbol = useMemo(() => {
+    if (!quote?.relayFee?.token) return;
+
+    return config.sdkConverter.findTokenConfigV1(
+      quote?.relayFee?.token,
+      Object.values(config.tokens),
+    )?.symbol;
+  }, [quote]);
+
+  useEffect(() => {
+    setCheckingBalance(true);
+    (async () => {
+      try {
+        if (!quote?.relayFee?.amount) return;
+
+        const wallet = getWalletConnection(TransferWallet.SENDING);
+        if (!wallet) return;
+
+        const cost = amount.whole(quote.relayFee.amount);
+
+        const balance = parseFloat(
+          await wallet.getBalance(quote.relayFee.token.address?.toString()),
+        );
+
+        if (typeof balance !== 'number') {
+          throw new Error(
+            `Expected balance to be a number, but got "${balance}"`,
+          );
+        }
+
+        if (isMounted.current) setState({ balance, cost });
```

Comment:
I don't think this is idiomatic React. you should return [cleanup code](https://react.dev/reference/react/useEffect#usage) in your `useEffect` instead of using this `isMounted` thing. set a boolean `canceled = false` and return a function from your hook that sets it to true. then just check for `if (!canceled)` here instead.

Reply:
https://github.com/wormhole-foundation/wormhole-connect/pull/2748/files#diff-b9861d794f4711dee7eb03464dbf080cd493cffe2c3b5ca46d4e0d8b43a01afd

Reply:
@artursapek have you had time to re-check this? ty

Reply:
I don't understand. what I'm saying is `useIsMounted` shouldn't even exist.

Reply:
```

  useEffect(() => {
    let isActive = true;

    setCheckingBalance(true);
    (async () => {
      try {
        if (!quote?.relayFee?.amount) return;

        const wallet = getWalletConnection(TransferWallet.SENDING);
        if (!wallet) return;

        const cost = amount.whole(quote.relayFee.amount);

        const balance = parseFloat(
          await wallet.getBalance(quote.relayFee.token.address?.toString())
        );

        if (typeof balance !== 'number') {
          throw new Error(
            `Expected balance to be a number, but got "${balance}"`
          );
        }

        if (isActive) setState({ balance, cost });
      } catch (e) {
        console.error(e);
        if (isActive) setState(null);
      } finally {
        if (isActive) setCheckingBalance(false);
      }
    })();

    return () => {
      isActive = false;
    };
  }, [quote]);
  ```

Reply:
this is mounted hook has that same logic to be reutilised

Reply:
pushed those changes